protocol/read-write-resource/read-method-support.feature

```diff
@@ -0,0 +1,48 @@
+Feature: Servers MUST support GET, HEAD and OPTIONS
+
+  Background: Set up container
+    * def testContainer = rootTestContainer.createContainer()
+    * def container = testContainer.createContainer()  
+    * def exampleTurtle = karate.readAsString('../fixtures/example.ttl')
+    * def rdfResource = testContainer.createResource('.ttl', exampleTurtle, 'text/turtle');
+    * def expected = parse(exampleTurtle, 'text/turtle')
+    
+  Scenario: GET is supported for containers
+    Given url testContainer.url
+    And headers clients.alice.getAuthHeaders('GET', testContainer.url)
+    When method GET
+    Then assert responseStatus != 400 && responseStatus <= 403
+
+  Scenario: GET is supported for resources
+    Given url rdfResource.url
+    And headers clients.alice.getAuthHeaders('GET', rdfResource.url)
+    When method GET
+    Then assert responseStatus != 400 && responseStatus <= 403
+
+  Scenario: HEAD is supported for containers
+    Given url testContainer.url
+    And headers clients.alice.getAuthHeaders('HEAD', testContainer.url)
+    When method HEAD
+    Then assert responseStatus != 400 && responseStatus <= 403
+
+  Scenario: HEAD is supported for resources
+    Given url rdfResource.url
+    And headers clients.alice.getAuthHeaders('HEAD', rdfResource.url)
+    When method HEAD
+    Then assert responseStatus != 400 && responseStatus <= 403
+
+
+  Scenario: OPTIONS is supported for containers
+    Given url testContainer.url
+    And headers clients.alice.getAuthHeaders('OPTIONS', testContainer.url)
+    When method OPTIONS
+    Then assert responseStatus != 400 && responseStatus <= 403
+
+  Scenario: OPTIONS is supported for resources
+    Given url rdfResource.url
+    And headers clients.alice.getAuthHeaders('OPTIONS', rdfResource.url)
+    When method OPTIONS
+    Then assert responseStatus != 400 && responseStatus <= 403
+
+    
+# TODO: What does requirement really mean? Since authn isn't required, should we test without authn and allow 40x, or should we always test with out and require a 200 response?
```

Comment:
Unless the server returns 405, it is deemed to be supported => the method is understood / allowed / supported.

Reply:
Indeed, I thought about that interpretation, but understood / allowed / supported can potentially mean different things. Allowed has a defined meaning in HTTP, if that is what we mean, then it is what we should write.

However, I'm not sure that is what we should mean, as it ties into all other functionality. It therefore seems that "support" is appropriate as that can allude to other requirements.